src/Microsoft.DotNet.Deployment.Tasks.Links/src/AkaMSLinksManager.cs

```diff
@@ -389,21 +395,49 @@ public async Task DeleteLinksAsync(List<string> linksToDelete)
 
         private async Task<HttpClient> CreateClient()
         {
+            //TODO remove feature switch: https://github.com/dotnet/arcade/issues/13318
+
+            if (_useIdentityClientLibrary)
+            {
+                _log.LogMessage(MessageImportance.High, "Creating a client using MSAL Library");
+                
+                if (_akamsLinksApp == null)
+                {
+                    _akamsLinksApp = ConfidentialClientApplicationBuilder.Create(_clientId)
+                    .WithClientSecret(_clientSecret)
+                    .WithAuthority(Authority)
+                    .Build();
+                }
+
+                Identity.Client.AuthenticationResult token = await _akamsLinksApp.AcquireTokenForClient(
+                    new[] { $"{Endpoint}/.default" })
+                    .WithTenantId(_tenant)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
```

Comment:
This indentation feels wrong
```suggestion
                Identity.Client.AuthenticationResult token = await _akamsLinksApp
                    .AcquireTokenForClient(new[] { $"{Endpoint}/.default" })
                    .WithTenantId(_tenant)
                    .ExecuteAsync()
                    .ConfigureAwait(false);
```